contracts/WitnetProxy.sol

```diff
@@ -0,0 +1,102 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.6;
+
+import "./utils/Upgradable.sol";
+
+/** @title WitnetProxy: upgradable delegate-proxy contract that routes Witnet data requests coming from a 
+ * `UsingWitnet`-inheriting contract to a currently active `WitnetRequestBoard` implementation. 
+ *
+ * https://github.com/witnet/witnet-ethereum-bridge/tree/0.3.x
+ *
+ * Written in 2021 by the Witnet Foundation.
+ **/
+contract WitnetProxy {
+  Upgradable public delegate;
+
+  /// @dev Constructor with no params as to ease eventual support of Singleton pattern (i.e. ERC-2470)
+  constructor () {}
+
+  /// @dev WitnetProxies will never accept direct transfer of ETHs.
+  receive() external payable {
+    revert("WitnetProxy: no ETH accepted");
+  }
+
+  /// @dev Payable fallback accepts delegating calls to payable functions.  
+  fallback() external payable { /* solhint-disable no-complex-fallback */
```

Comment:
When possible, use `solhint-disable-next-line` instead, I believe `solhint-disable` disables the lints for the rest of the current file:

```suggestion
  // solhint-disable-next-line no-complex-fallback
  fallback() external payable {
```